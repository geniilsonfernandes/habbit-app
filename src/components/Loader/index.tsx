import React from 'react'
import styled, { css, keyframes } from 'styled-components'

export type LoaderProps = {
  size?: 'small' | 'medium' | 'large'
}

const LoadingAnimation = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`

const Loading = styled.div<LoaderProps>`
  ${({ theme, size = 'medium' }) => css`
    border: 0.4rem solid rgba(253, 253, 253, 0.096);
    border-top: 0.4rem solid ${theme.colors.primary[300]};
    border-radius: 50%;
    width: ${size === 'small' ? '2rem' : size === 'medium' ? '2rem' : '3rem'};
    height: ${size === 'small' ? '2rem' : size === 'medium' ? '2rem' : '3rem'};
    animation: ${LoadingAnimation} 0.6s linear infinite;
  `}
`

const Loader = (props: LoaderProps) => {
  return <Loading {...props} aria-label="Loading" />
}

export default Loader
