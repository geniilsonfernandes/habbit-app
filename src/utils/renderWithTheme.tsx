import { ThemeProvider } from 'styled-components'
import { RenderResult, render } from '@testing-library/react'
import theme from '../styles/theme'

export const renderWithTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children} </ThemeProvider>)
}

export default renderWithTheme
