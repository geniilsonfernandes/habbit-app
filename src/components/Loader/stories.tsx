import { Story, Meta } from '@storybook/react'

import Loader, { LoaderProps } from '.'

export default {
  component: Loader,
  title: 'Loader',
} as Meta<LoaderProps>

export const Basic: Story<LoaderProps> = (args) => <Loader {...args} />
