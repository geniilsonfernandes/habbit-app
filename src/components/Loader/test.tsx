import { screen } from '@testing-library/react'
import Loader from '.'
import renderWithTheme from '../../utils/renderWithTheme'

describe('<Loader />', () => {
  it('should render the loader', () => {
    renderWithTheme(<Loader />)

    const loader = screen.getByLabelText(/loading/i)

    expect(loader).toBeInTheDocument()
    expect(loader).toMatchSnapshot()
  })

  it('should render the loader with size small', () => {
    renderWithTheme(<Loader size="small" />)

    const loader = screen.getByLabelText(/loading/i)

    expect(loader).toBeInTheDocument()
    expect(loader).toHaveStyle({ width: '2rem' })
  })

  it('should render the loader with size medium', () => {
    renderWithTheme(<Loader size="medium" />)

    const loader = screen.getByLabelText(/loading/i)

    expect(loader).toBeInTheDocument()
    expect(loader).toHaveStyle({ width: '2rem' })
  })

  it('should render the loader with size large', () => {
    renderWithTheme(<Loader size="large" />)

    const loader = screen.getByLabelText(/loading/i)

    expect(loader).toBeInTheDocument()
    expect(loader).toHaveStyle({ width: '3rem' })
  })
})
