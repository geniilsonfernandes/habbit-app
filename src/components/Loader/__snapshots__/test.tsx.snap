// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render the loader 1`] = `
.c0 {
  border: 0.4rem solid rgba(253,253,253,0.096);
  border-top: 0.4rem solid #9DA2F9;
  border-radius: 50%;
  width: 2rem;
  height: 2rem;
  -webkit-animation: fEWCgj 0.6s linear infinite;
  animation: fEWCgj 0.6s linear infinite;
}

<div
  aria-label="Loading"
  class="c0"
/>
`;
